src/collective/recaptcha/testing.py

```diff
@@ -1,14 +1,30 @@
-# encoding: utf-8
-from plone.app.testing import PLONE_INTEGRATION_TESTING
-from plone.app.testing import PloneWithPackageLayer
+# -*- coding: utf-8 -*-
+from plone.app.testing import applyProfile
+from plone.app.testing import IntegrationTesting
+from plone.app.testing import PLONE_FIXTURE
+from plone.app.testing import PloneSandboxLayer
 
 import collective.recaptcha
 
 
-COLLECTIVE_RECAPTCHA = PloneWithPackageLayer(
-    zcml_filename="configure.zcml",
-    zcml_package=collective.recaptcha,
-    additional_z2_products=(),
-    bases=(PLONE_INTEGRATION_TESTING,),
-    name="COLLECTIVE_RECAPTCHA",
+class CollectiveRecaptchaLayer(PloneSandboxLayer):
+
+    defaultBases = (PLONE_FIXTURE,)
+
+    def setUpZope(self, app, configurationContext):  # @UnusedVariable
+        # Load any other ZCML that is required for your tests.
+        # The z3c.autoinclude feature is disabled in the Plone fixture base
+        # layer.
+        self.loadZCML(package=collective.recaptcha)
+
+    def setUpPloneSite(self, portal):
+        applyProfile(portal, "collective.recaptcha:default")
+
+
+COLLECTIVE_RECAPTCHA_FIXTURE = CollectiveRecaptchaLayer()
+
+
+COLLECTIVE_RECAPTCHA_INTEGRATION_TESTING = IntegrationTesting(
```

Comment:
I think with the original could this can be done like:

```
COLLECTIVE_RECAPTCHA_INTEGRATION_TESTING = IntegrationTesting(
    bases=(COLLECTIVE_RECAPTCHA,),
     name="CollectiveRecaptchaLayer:IntegrationTesting",
 )
```

The `PloneWithPackageLayer` is quite handy, although I confess I hardly ever remember to use it. :-)

Reply:
Well I used `PloneSandboxLayer` because the vast majority of packages use it.

I saw that in the current code the `gs_profile_id parameter` is not passed in the `PloneWithPackageLayer` constructor. I must add this parameter, correct? See:

https://github.com/plone/plone.app.testing/blob/37f569239bc49fb69106ae115fc1c55c7f83d4e8/src/plone/app/testing/helpers.py#L507-L520

Should we encourage the use of `PloneWithPackageLayer`? Should we replace `PloneSandboxLayer` with `PloneWithPackageLayer` in `bobtemplates.plone`?

Reply:
Yes, you would need to pass `gs_profile_id`. Strange that this was not done yet.

Using `PloneWithPackageLayer` in `bobtemplates.plone` sounds good yes.

Reply:
Done.